dlt/destinations/job_client_impl.py

```diff
@@ -573,16 +646,29 @@ def _norm_and_escape_columns(self, *columns: str) -> Iterator[str]:
         )
 
     def get_stored_schema_by_hash(self, version_hash: str) -> StorageSchemaInfo:
-        table_name = self.sql_client.make_qualified_table_name(self.schema.version_table_name)
-        (c_version_hash,) = self._norm_and_escape_columns("version_hash")
+        table_name = self.schema.version_table_name
+        c_version_hash = self.schema.naming.normalize_path("version_hash")
 
-        maybe_limit_clause_1, maybe_limit_clause_2 = self.sql_client._limit_clause_sql(1)
+        select_expr = build_select_expr(
+            table_name=table_name,
+            selected_columns=list(self.version_table_schema_columns.keys()),
+            quoted_identifiers=True,
+        )
 
-        query = (
-            f"SELECT {maybe_limit_clause_1} {self.version_table_schema_columns} FROM"
-            f" {table_name} WHERE {c_version_hash} = %s {maybe_limit_clause_2};"
+        where_condition = sqlglot.exp.EQ(
+            this=sqlglot.exp.Column(this=sqlglot.exp.Identifier(this=c_version_hash, quoted=True)),
+            expression=sqlglot.exp.Literal.string(version_hash),
         )
-        return self._row_to_schema_info(query, version_hash)
+
+        select_expr = select_expr.where(where_condition).limit(1)
+
+        dialect = self.sql_client.capabilities.sqlglot_dialect
```

Comment:
this whole code block is repeated 4 times, you could create a helper function on the jobclient that renders a sqlglot query object into the the current dialect.